assembl/static2/js/app/pages/debate.jsx

```diff
@@ -30,11 +34,41 @@ class Debate extends React.Component {
   displayThumbnails() {
     this.setState({ isThumbnailsHidden: !this.state.isThumbnailsHidden });
   }
+  redirectToV1Threads() {
+    const { debateData } = this.props.debate;
+    const { identifier } = this.props;
+    const locale = this.props.lang;
+    const phaseName = getPhaseName(debateData.timeline, identifier, locale);
+    const body = <Translate value="debate.redirectToThreadPhase" phaseName={phaseName} />;
+    displayModal(null, body, true, null, null, true);
+    const slug = { slug: debateData.slug };
+    const redirectionURL = get('oldDebate', slug);
+    setTimeout(() => {
+      location.href = redirectionURL;
+    }, 6000);
+  }
   render() {
+    const { debateData } = this.props.debate;
+    const { identifier } = this.props; // identifier of current phase being accessed
     const { loading, thematics } = this.props.data;
-    const { identifier, isNavbarHidden } = this.props;
+    const { isNavbarHidden } = this.props;
     const isParentRoute = !this.props.params.phase || false;
     const themeId = this.props.params.themeId || null;
+
+    if (identifier && !isPhaseStarted(debateData.timeline, identifier)) {
```

Comment:
I think I already tried the componentDidMount way and obtained the same warnings/errors from React. I would be happy to do a quick pair programming session with one of you about this :)